scripts/azureml-assets/azureml/assets/release/asset_publish.py

```diff
@@ -297,14 +297,27 @@ def _str2bool(v: str) -> bool:
                             run_command(cmd, failure_list, debug_mode)
                 except Exception as e:
                     logger.log_error(f"Exception in loading model config: {str(e)}")
-                    failure_list.append(asset.name)
+                    failure_list.append(asset)
 
             else:
                 logger.log_warning(f"unsupported asset type: {asset.type.value}")
 
     if len(failure_list) > 0:
-        logger.log_warning(
-            f"following assets failed to publish: {failure_list}")
+        failed_assets = defaultdict(list)
+        logger.print(f"failure_list: {failure_list}")
+        for asset in failure_list:
+            logger.print(f"processing failed asset: {asset.name}")
+            failed_assets[asset.type.value].append(asset.name)
+
+        logger.print(f"failed assets: {failed_assets}")
+        for asset_type, asset_names in failed_assets.items():
+            logger.log_warning(f"Failed to register {asset_type}s: {asset_names}")
+        # the following dump process will generate a yaml file for the report
+        # process in the end of the publishing script
+        result_path = Path().resolve() / "failed_assets.yml"
```

Comment:
Ideally this filename should come from an arg passed to the script.

Reply:
done